# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/TouchManager.cpp
# Commit: 5718f7f3067e
# Full Hash: 5718f7f3067eff42e097a6013acd82fbe6aa299a
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2025-05-15 22:58:00
# Regressor Bug: 1550462
# File Overlap Count: 1
# Description:
#   Bug 1550462 - part 2: Make `PresShell::HandleEvent` dispatch preceding `pointerrawupdate` event r=smaug,dom-core,edgar
#   
#   This patch tries to dispatch `ePointerRawUpdate` with
#   `PresShell::EventHandler::DispatchPrecedingPointerEvent` as same as usual
#   pointer events.
# ==============================================================================

diff -r aae164cff6aa -r 5718f7f3067e layout/base/TouchManager.cpp
--- a/layout/base/TouchManager.cpp	Thu May 15 03:20:28 2025 +0000
+++ b/layout/base/TouchManager.cpp	Thu May 15 03:20:29 2025 +0000
@@ -303,6 +303,9 @@
       }
       break;
     }
+    case eTouchRawUpdate:
+      MOZ_ASSERT_UNREACHABLE("eTouchRawUpdate shouldn't be handled as a touch");
+      break;
     case eTouchMove: {
       // Check for touches that changed. Mark them add to queue
       WidgetTouchEvent* touchEvent = aEvent->AsTouchEvent();
@@ -448,6 +451,9 @@
 void TouchManager::PostHandleEvent(const WidgetEvent* aEvent,
                                    const nsEventStatus* aStatus) {
   switch (aEvent->mMessage) {
+    case eTouchRawUpdate:
+      MOZ_ASSERT_UNREACHABLE("eTouchRawUpdate shouldn't be handled as a touch");
+      break;
     case eTouchMove: {
       if (sSingleTouchStartTimeStamp.IsNull()) {
         break;
@@ -559,7 +565,8 @@
       // We don't want to fire duplicated pointerdown.
       return false;
     }
-    case eTouchMove: {
+    case eTouchMove:
+    case eTouchRawUpdate: {
       return !aTouch->Equals(info.mTouch);
     }
     default: